### DISCUSSION TITLE
ROCm-Device-Libs Compile error.  invalid value 'dynamic' in '-fdenormal-fp-math=dynamic'

### DISCUSSION THREAD

**Discussion by rohezal** [ROCm/ROCm#2650] [Q&A] [answered]
Ubuntu 22.04 on AMD Threadripper 1800X with 6700 XT here.
It does not matter if I compile this package with clang or gcc, there is always an the same error, error: invalid value 'dynamic' in '-fdenormal-fp-math=dynamic'. Looks like a llvm problem.
 .
```

 cmake ..
-- The C compiler identification is Clang 14.0.0
-- The CXX compiler identification is Clang 14.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test Terminfo_LINKABLE
-- Performing Test Terminfo_LINKABLE - Success
-- Found Terminfo: /usr/lib/x86_64-linux-gnu/libtinfo.so  
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11")  
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.13") 
-- Linker detection: GNU ld
CMake Deprecation Warning at cmake/OCL.cmake:12 (cmake_policy):
  The OLD behavior for policy CMP0053 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  CMakeLists.txt:60 (include)


-- FileCheck not found, not adding constant fold tests
device-libs CPACK_RPM_PACKAGE_RELEASE now is local
-- Configuring done (1.5s)
-- Generating done (0.1s)

```

```
make 
[  0%] Building CXX object utils/prepare-builtins/CMakeFiles/prepare-builtins.dir/prepare-builtins.cpp.o
[  1%] Linking CXX executable prepare-builtins
[  1%] Built target prepare-builtins
[  2%] Generating abi_version_400.bc
error: invalid value 'dynamic' in '-fdenormal-fp-math=dynamic'
make[2]: *** [oclc/CMakeFiles/oclc_abi_version_400.dir/build.make:73: oclc/abi_version_400.bc] Error 1
make[1]: *** [CMakeFiles/Makefile2:491: oclc/CMakeFiles/oclc_abi_version_400.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

```

**Comment by sohaibnd** [comment, marked as answer]
Hi @rohezal, the ROCm-Device-Libs repo has been moved under the `amd/device-libs` directory in the [ROCm/llvm-project](https://github.com/ROCm/llvm-project) repo (which is the AMD fork of the LLVM Project). Can you try following the instructions [here](https://github.com/ROCm/llvm-project/blob/amd-staging/amd/device-libs/README.md#building) to build the device libraries? I was able to build it without hitting any errors.

Note that the device libraries can be built either as a standalone project or as an llvm external subproject. It seems like you were building it as a standalone project, in which case you have to include the directory of your llvm build in CMAKE_PREFIX_PATH  (which requires building llvm first) when calling cmake: `cmake -DCMAKE_PREFIX_PATH=$LLVM_BUILD ..`

Please let me know if that fixes you issue!

**Comment by b-sumner** [comment]
@rohezal the device library and compiler are a matched set.  That is why the sources were moved.  If you're using your own hash of LLVM, then you should probably use a hash of device libraries matching the date of your LLVM hash.

**Comment by sohaibnd** [comment]
@rohezal I'm marking the above response as answer. If you're still having issues building the device libraries, please update this discussion and we can look into it further.
